library/pod-security-policy/selinux/template.yaml

```diff
@@ -56,89 +56,147 @@ spec:
                     description: "An SELinux user."
   targets:
     - target: admission.k8s.gatekeeper.sh
-      rego: |
-        package k8spspselinux
+      code:
+      - engine: K8sNativeValidation
+        source:
+          variables:
+          - name: usingAllowedSELinuxOptions
+            expression: |
+              has(variables.anyObject.spec.securityContext) && has(variables.anyObject.spec.securityContext.seLinuxOptions) ? 
+                (has(variables.params.allowedSELinuxOptions) ? 
+                (
+                  size(variables.params.allowedSELinuxOptions) > 0 && variables.params.allowedSELinuxOptions.all(opts,
+                    (has(opts.level) ? has(variables.anyObject.spec.securityContext.seLinuxOptions.level) && (variables.anyObject.spec.securityContext.seLinuxOptions.level == opts.level) : !has(variables.anyObject.spec.securityContext.seLinuxOptions.level)) &&
+                    (has(opts.role) ? has(variables.anyObject.spec.securityContext.seLinuxOptions.role) && (variables.anyObject.spec.securityContext.seLinuxOptions.role == opts.role) : !has(variables.anyObject.spec.securityContext.seLinuxOptions.role)) &&
+                    (has(opts.type) ? has(variables.anyObject.spec.securityContext.seLinuxOptions.type) && (variables.anyObject.spec.securityContext.seLinuxOptions.type == opts.type) : !has(variables.anyObject.spec.securityContext.seLinuxOptions.type)) &&
+                    (has(opts.user) ? has(variables.anyObject.spec.securityContext.seLinuxOptions.user) && (variables.anyObject.spec.securityContext.seLinuxOptions.user == opts.user) : !has(variables.anyObject.spec.securityContext.seLinuxOptions.user))
+                  )
+                ) :  
+                true) 
+                : true
+          - name: containers
```

Comment:
I have not looked at containers code, but I'm guessing it will need to be modified to comply with pod level code.